### ISSUE TITLE
Probable problem with paths, various things 'not found' (`util/util')

### ISSUE THREAD

**Issue by adaminfinitum** [foundation/foundation-sites#12279]
## What should happen?
I'm customizing a clone of this repo (v6.7.1) and have installed all the dependencies so it should compile the SCSS to CSS, without errors.
...

## What happens instead?
My bash shell console (Terminal) throws errors during compilation.
...

## Possible Solution

Given the number of times a similar issue has been submitted over the years, I think an "explain like I'm five" style tutorial and explanation might be in order.

If I recall correctly, Foundation 5.4.3 did not require me to uncomment a line (specifically, line 21 in `foundation.scss` which is `@import 'settings/settings';`) to make my customizations to the settings compile, but this version did seem to require that.

...

## Test Case and/or Steps to Reproduce (for bugs)

Test Case: 
The codepen.io link isn't to a current version of Foundation it is to 6.5.1 and this is something that I can't reproduce in a single HTML file and a single (S)CSS file.

There's a whole bunch of files, that just aren't working together correctly in this instance (and none of the existing issues give clear guidance on how to fix it).

Does CodePen support git, maybe if you are a paying user (I am not) but the term `fork` confuses me...unless they do support git in which case, I just don't have access to those features.

How to reproduce:
1. I don't rightly know. I haven't made major changes, and I use Sublime Text 3 with syntax highlighting and linting so I don't think my code is incorrect.
2. I simply don't know why I'm seeing errors about `util/util` being not found or unreadable...and it seems a lot of people over the years have had the same problem.
3.

## Context

I am customizing the settings for an app that I am working on front-end development for so I made minor changes to `settings/_settings.scss` and added some code (I like the gradient buttons from Twitter Bootstrap 2.x) so added the code from the official Sass port of TWBS 2.3.2.2.

TWBS 2.x uses `camelCase` naming conventions for variables so `$baseLineHeight` needs to be replaced with `$global-lineheight` etc., so I completely expected there to be some errors that I'd fix as they came up.

Before I started trying to implement code from TWBS, the only customizations I had made were to the `settings/_settings.scss` file (changing the values of the color schemes to match my needs, customizing the `$button-radius` etc.)

Even before implementing any custom code (yes, I had customized the `_settings.scss` file, but I hadn't implemented any of the Bootstrap 2 SCSS code) I got the following error:
```
Message:
    scss/settings/_settings.scss
Error: File to import not found or unreadable: util/util.
        on line 63 of scss/settings/_settings.scss
        from line 22 of scss/foundation.scss
        from line 3 of assets/foundation-float.scss
>> @import 'util/util';
```
Initially I ignored it, but after I fixed all the variables and conflicts with TWBS variables, the error returned.

Some of the prior issues and docs that I read trying to figure out how to fix it are:

[https://github.com/foundation/foundation-sites/issues/10537](https://github.com/foundation/foundation-sites/issues/10537)
[https://github.com/foundation/foundation-sites/issues/8293](https://github.com/foundation/foundation-sites/issues/8293)
[https://stackoverflow.com/questions/40694325/sass-stops-compiling-file-to-import-not-found-or-unreadable-util-util](https://stackoverflow.com/questions/40694325/sass-stops-compiling-file-to-import-not-found-or-unreadable-util-util)
and
[https://get.foundation/sites/docs/sass.html#the-settings-file](https://get.foundation/sites/docs/sass.html#the-settings-file)

They seem to be on topic, but I can't really find a cohesive answer in any of them.

One of them says I need to copy and move my `settings/_settings.scss` file to my project directory but I don't see how that could be correct (it's in my project already).

Another suggested changing `@import 'util/util';` to `@import '../util/util';` so I tried that and now variables which are defined in my `settings/_settings.scss` file are triggering errors such as `Error: Undefined variable: "$global-lineheight".` 
...

## Your Environment

- Foundation version(s) used: 6.7.1
- Browser(s) name and version(s): (not a browser related issue but) GNU bash, version 5.1.4(1)-release (x86_64-apple-darwin19.6.0). Node 14.17.4, NPM 7.20.3 (no native Sass installed).

`package.json`:
```
{
  "name": "foundation-sites",
  "version": "6.7.1",
  "main": "dist/js/foundation.js",
  "module": "dist/js/foundation.esm.js",
  "jsnext:main": "dist/js/foundation.esm.js",
  "esnext": "dist/js/foundation.es6.js",
  "typings": "dist/js/foundation.d.ts",
  "description": "The most advanced responsive front-end framework in the world.",
  "author": "Foundation Yetinauts <contact@get.foundation> (https://get.foundation)",
  "homepage": "https://get.foundation/sites",
  "scripts": {
    "start": "gulp",
    "test": "yarn test:sass && yarn test:javascript:units && yarn test:bundle",
    "test:single-process": "yarn test:sass && yarn test:javascript:units:single-process && yarn test:bundle:single-process",
    "test:ci": "yarn test:sass && yarn test:javascript:ci",
    "test:ci:single-process": "yarn test:sass && yarn test:javascript:ci:single-process",
    "test:sass": "mocha test/sass/test_sass.js --colors",
    "test:javascript:transpile": "gulp sass:foundation --color && gulp test:transpile-js --color",
    "test:javascript:units": "yarn test:javascript:transpile && mocha-headless-chrome -a ignore-resource-errors -a no-sandbox -a disable-setuid-sandbox -a disable-dev-shm-usage -f test/javascript/index.html",
    "test:javascript:units:single-process": "yarn test:javascript:transpile && mocha-headless-chrome -a single-process -a ignore-resource-errors -a no-sandbox -a disable-setuid-sandbox -a disable-dev-shm-usage -f test/javascript/index.html",
    "test:javascript:browserstack": "yarn test:javascript:transpile && browserstack-runner",
    "test:javascript:ci": "yarn test:javascript:transpile && mocha-headless-chrome -a ignore-resource-errors -a no-sandbox -a disable-setuid-sandbox -a disable-dev-shm-usage -f test/javascript/index.html && browserstack-runner",
    "test:javascript:ci:single-process": "yarn test:javascript:transpile && mocha-headless-chrome -a single-process -a ignore-resource-errors -a no-sandbox -a disable-setuid-sandbox -a disable-dev-shm-usage -f test/javascript/index.html && browserstack-runner",
    "test:bundle": "mocha-headless-chrome -a ignore-resource-errors -a no-sandbox -a disable-setuid-sandbox -a disable-dev-shm-usage -f test/bundle/index.html",
    "test:bundle:single-process": "mocha-headless-chrome -a ignore-resource-errors -a single-process -a no-sandbox -a disable-setuid-sandbox -a disable-dev-shm-usage -f test/bundle/index.html",
    "test:visual": "gulp test --color",
    "deploy": "gulp deploy --color",
    "deploy:prep": "gulp deploy:prep --color"
  },
  "peerDependencies": {
    "jquery": ">=2.2.0",
    "what-input": ">=4.1.0"
  },
  "license": "MIT",
  "devDependencies": {
    "@babel/core": "^7.2.2",
    "@babel/preset-env": "^7.2.3",
    "@commitlint/cli": "^8.1.0",
    "@commitlint/config-conventional": "^7.3.1",
    "autoprefixer": "^9.4.5",
    "babel-loader": "^8.0.5",
    "browser-sync": "^2.18.12",
    "browserstack-runner": "^0.9.0",
    "chai": "^4.1.2",
    "chai-jquery": "^2.1.0",
    "clipboard": "^2.0.4",
    "corejs-typeahead": "^1.1.1",
    "cz-conventional-changelog": "^2.1.0",
    "doiuse": "^4.4.1",
    "foundation-docs": "github:foundation/foundation-docs",
    "gulp": "^4.0.0",
    "gulp-add-src": "^1.0.0",
    "gulp-babel": "^8.0.0",
    "gulp-cache-bust": "^1.3.0",
    "gulp-check-deps": "^1.4.1",
    "gulp-clean-css": "^4.0.0",
    "gulp-concat": "^2.4.3",
    "gulp-eslint": "^5.0.0",
    "gulp-filter": "^5.1.0",
    "gulp-if": "^2.0.0",
    "gulp-load-plugins": "^1.5.0",
    "gulp-newer": "^1.1.0",
    "gulp-plumber": "^1.2.1",
    "gulp-postcss": "^8.0.0",
    "gulp-rename": "^1.2.2",
    "gulp-replace": "^1.0.0",
    "gulp-sass": "^4.1.1",
    "gulp-sass-lint": "^1.4.0",
    "gulp-sourcemaps": "^2.6.4",
    "gulp-uglify": "^3.0.0",
    "gulp-zip": "^4.1.0",
    "husky": "^1.3.1",
    "is-empty-object": "^1.1.1",
    "jquery": "^3.3.1",
    "js-yaml": "^3.12.1",
    "kleur": "^3.0.2",
    "mocha": "^5.0.5",
    "mocha-headless-chrome": "^2.0.1",
    "motion-ui": "^2.0.2",
    "node-sass": "^6.0.1",
    "octophant": "^1.0.0",
    "panini": "^1.7.2",
    "patch-package": "^6.2.1",
    "prettyjson": "^1.1.3",
    "puppeteer": "^1.11.0",
    "require-dir": "^1.2.0",
    "requirejs": "^2.3.6",
    "rollup": "^1.1.0",
    "rollup-plugin-babel": "^4.3.0",
    "run-sequence": "^2.2.1",
    "sass-true": "^4.0.0",
    "sassy-lists": "^3.0.1",
    "sinon": "^7.2.2",
    "strip-indent": "^2.0.0",
    "supercollider": "^1.5.3",
    "touch": "^3.1.0",
    "undertaker-forward-reference": "^1.0.2",
    "util-deprecate": "^1.0.2",
    "vinyl": "^2.0.2",
    "vinyl-named": "^1.1.0",
    "webpack": "^4.28.4",
    "webpack-stream": "^5.2.1",
    "what-input": ">=4.1.0",
    "yargs": "^13.3.0"
  },
  "resolutions": {
    "puppeteer": "^1.11.0",
    "glob-parent": "^5.1.2",
    "marked": "^2",
    "node-sass": "^6.0.1",
    "lodash": "^4.17.21",
    "minimist": "^1.2.3",
    "dot-prop": "^5.2.0",
    "trim-newlines": "^3.0.1"
  },
  "engines": {
    "node": ">=12.0"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/foundation/foundation-sites"
  },
  "bugs": {
    "url": "https://github.com/foundation/foundation-sites/issues",
    "email": "contact@get.foundation"
  },
  "keywords": [
    "css",
    "scss",
    "sass",
    "js",
    "responsive",
    "mobile-first",
    "framework",
    "eyeglass-module",
    "handlebars-helper-rel",
    "handlebars-helper-slugify"
  ],
  "eyeglass": {
    "name": "foundation",
    "sassDir": "scss",
    "needs": ">=0.8.0",
    "exports": false
  },
  "jspm": {
    "main": "dist/js/foundation.esm.js",
    "format": "esm"
  },
  "browserify": {
    "transform": [
      [
        "babelify",
        {
          "presets": [
            "@babel/preset-env"
          ]
        }
      ]
    ]
  },
  "config": {
    "commitizen": {
      "path": "./node_modules/cz-conventional-changelog"
    }
  }
}

```
- Device, Operating System and version: MacBook Pro, MacOS 10.15.7
- Link to your project: private (If needed, I could create a public repo on github with only the files that have been modified from 6.7.1).

## Checklist

- [x] I have read and follow the CONTRIBUTING.md document.
- [x] There are no other issues similar to this one.
- [x] The issue title and template are correctly filled.

I believe I have complied...there are similar issues, perhaps half a dozen that I read (none that I saw were still open) and I still don't know what is wrong.

And it seems there has been significant confusion about this for a long time so some current clarifying documentation would be really helpful.

**Comment by adaminfinitum** [comment]
I don't really understand this, but through trial and error, it seems to be compiling correctly with modifications to:
(the "root" directory of the cloned repo is "foundation-sites" which for the purposes of this code I will consider to be `/`, so my settings file is located at `/scss/settings/_settings.scss`)

On line 63 of `/scss/settings/_settings.scss` I changed `@import 'util/util';` to `@import 'scss/util/util';`
On line 21 of `/scss/foundation.scss` I changed `@import 'settings/settings';` to `@import 'scss/settings/settings';`

Since the error mentioned line 3 of `/assets/foundation-float.scss` which is `@import '../scss/foundation';`, I tried a few different variations of the paths in that file such as:
- `@import '/scss/foundation';`
- `@import 'foundation-sites/scss/foundation';`
- `@import 'scss/foundation';`

As long as the `_settings.scss` file and the `foundation.scss` file contain the the modifications above, Sass would compile without an error with either of: 
- `@import 'scss/foundation';`
- `@import '../scss/foundation';` (the original path, in the unedited file, which makes sense because I never modified it and was puzzled when it was mentioned in the error message).

From `/assets/foundation-float.scss` the correct path would be `../scss/foundation` (up one directory, then down into the `scss` directory to the file `foundation.scss`).

I really don't understand why both `../scss/foundation` and `scss/foundation` work.

I also don't understand why the paths in `/scss/settings/_settings.scss` would need to be modified to `@import 'scss/util/util';` ... that's a relative path so wouldn't I be telling Sass to look for `/scss/settings/scss/util/util` (which is actually `/scss/settings/scss/util/_util.scss` ?

If the "foundation-sites" directory is the functional equivalent (within the repo) of me entering `cd /` into my bash shell, then why doesn't  `@import '/scss/foundation';` work in my `/assets/foundation-floats.scss` file?

I'm guessing this is some quirk of `Sass` or maybe of `node-sass` or `gulp-sass` or `sass-true`, but I don't understand how or why the various examples above either work or don't work.
